rdmd.d

```diff
@@ -526,39 +627,89 @@ private int rebuild(string root, string fullExe,
-            remove(fullExeTemp);
-        return result;
-    }
-    // clean up the dir containing the object file, just not in dry
-    // run mode because we haven't created any!
-    if (!dryRun)
+    auto fullCommand = [ compiler ] ~ todo;
+    if (dryRun)
+        yap("spawn ", fullCommand);
+    else
     {
-        yap("stat ", objDir);
-        if (objDir.exists && objDir.startsWith(workDir))
+        Pipe stderrPipe;
+        File stderrArg;
+        if (useDashI)
+        {
+            stderrPipe = pipe();
+            stderrArg = stderrPipe.writeEnd;
+        }
+        else
         {
-            yap("rmdirRecurse ", objDir);
-            // We swallow the exception because of a potential race: two
-            // concurrently-running scripts may attempt to remove this
-            // directory. One will fail.
-            collectException(rmdirRecurse(objDir));
+            stderrArg = stderr;
+        }
+        auto exitCode = runProcess(fullCommand, stdin, stdout, stderrArg);
+        if (exitCode)
+        {
+            yap("stat ", fullExeTemp);
+            if (exists(fullExeTemp))
+            {
+               yap("remove ", fullExeTemp);
+               remove(fullExeTemp);
+            }
+            yap("stat ", depsFilename);
+            if(exists(depsFilename))
+            {
+                yap("remove ", depsFilename);
+                remove(depsFilename);
+            }
+            if (useDashI)
+            {
+                auto firstLine = stderrPipe.readEnd.readln();
+                if(firstLine.canFind("unrecognized switch '-i'"))
+                {
+                    return BuildResult(Yes.dashIError);
+                }
+                stderr.writeln(firstLine);
+            }
         }
-        yap("mv ", fullExeTemp, " ", fullExe);
-        rename(fullExeTemp, fullExe);
+        if (useDashI)
+        {
+           // dump rest of stderr
+           foreach (line; stderrPipe.readEnd.byLine)
+           {
+               stderr.writeln(line);
+           }
+       }
+       if (exitCode)
+           return BuildResult(No.dashIError, exitCode);
+
+       // clean up the dir containing the object file
+       yap("stat ", objDir);
+       if (objDir.exists && objDir.startsWith(workDir))
+       {
+           yap("rmdirRecurse ", objDir);
+           // We swallow the exception because of a potential race: two
+           // concurrently-running scripts may attempt to remove this
+           // directory. One will fail.
+           collectException(rmdirRecurse(objDir));
+       }
+       if (!suppressOutput)
+       {
+           yap("mv ", fullExeTemp, " ", fullExe);
+           rename(fullExeTemp, fullExe);
+       }
     }
-    return 0;
+    return BuildResult(No.dashIError, 0);
 }
 
 // Run a program optionally writing the command line first
 // If "replace" is true and the OS supports it, replace the current process.
 
+private int runProcess(T...)(string[] args, T rest)
```

Comment:
* comment above refers to `run`, not `runProcess` (at least it mentions an arg `replace` that isn't in `runProcess`